### PR TITLE
Change default MySQL client to MariaDB

### PR DESCRIPTION
**Pull request by potiuk**
This PR is a response to pretty catastrophic issue caused by expiring key on MySQL repository on 14th of December. Oracle does not follow the best practices for signing their packages (while all others do) and their packages and repositories are signed with a key with short expiry date. This basically puts an expiry on their repository, and anyone who releases images following best practices of installation, while keeping the repository after installing mysql libraries has to rebuild their past released images every 2 years or so.

This is the last straw for our MySQL client installation problems (we had a few, especially for ARM images) and we decided that we will switch to MariaDB client libraries by default (while allowing our users to build custom images with MySQL libraries).

The issue tracked in Airflow repository is: #36231

The issues in Oracle's MySQL repo:

* https://bugs.mysql.com/bug.php?id=113427
* https://bugs.mysql.com/bug.php?id=113428
* https://bugs.mysql.com/bug.php?id=113432

This PR implements a number of changes:

* MariaDB client is now default client used for both ARM and X86
* both pre-2023 and 2023 keys for MySQL are now added to be trusted when custom image with MySQL client is built
* MySQL repository is removed after installing MySQL (to avoid repeating similar fiasco for MySQL users in 2025
* changelog added and instructions on how to build custom image with MySQL client
* one of our test suites is converted to use "current" image not latest released image (that was bug in our CI).
* test was added in `canary` and `release` builds in CI to also test build of custom image with MySQL client

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @ephraimbuddy - this one also contains  fix (:crossed_fingers: ) for the problem that we saw that one of our tests for extending images used the released version of Airflow image rather than the one built in the PR

File: scripts/docker/install_mysql.sh [hunk 1]
```diff
@@ -75,7 +75,8 @@ install_mysql_client() {
         exit 1
     fi
 
-    export_key "467B942D3A79BD29" "mysql"
+    export_key "467B942D3A79BD29" "mysql-2022"
```
Comment (Taragolis) [inline scripts/docker/install_mysql.sh]: I think it is no reason to import expired key, we can't use it legitimately without disable validation at all

Comment (potiuk) [reply scripts/docker/install_mysql.sh]: True.

**Comment by potiuk** [comment]
I also mixed `INSTALL_MYSQL_CLIENT` with `INSTALL_MYSQL_CLIENT_TYPE` in the initial version - corrected now.

The image is building nicely with `--install-mysql-client-type mysql`

File: scripts/docker/install_mysql.sh (hunk 1 above)
Comment (potiuk) [reply scripts/docker/install_mysql.sh]: Fixed

**Review by Taragolis — APPROVED** [review]
Looks good, just one place also need to be changed - Docker Arguments reference

https://github.com/apache/airflow/blob/c9647019deab91376fd7a10632cb1346daa53583/docs/docker-stack/build-arg-ref.rst?plain=1#L173

**Comment by potiuk** [comment]
> Looks good, just one place also need to be changed - Docker Arguments reference

Yep. Fixed. I corrected it but in generated rst.txt version (:facepalm: )

**Review by hussein-awala — APPROVED** [review]
Looks good

**Comment by potiuk** [comment]
Turned out that the fix in "prod" image detected an issue that openjdk needs higher version on bokwworm :) . Nice side-effect

**Review by ephraimbuddy — APPROVED** [review]

**Review by aritra24 — APPROVED** [review]

**Comment by potiuk** [comment]
And merged

**Pull request merged**